core/src/main/scala/org/apache/spark/deploy/rest/RestClient.scala

```diff
@@ -219,14 +221,20 @@ private[deploy] class StandaloneRestClient extends Logging {
 
   /** Return the base URL for communicating with the server, including the protocol version. */
   private def getBaseUrl(master: String): String = {
-    val masterUrl = master.stripPrefix("spark://").stripSuffix("/")
+    val masterUrl = if (master.startsWith("spark://")) {
+      master.stripPrefix("spark://").stripSuffix("/")
+    } else {
+      master.stripPrefix("mesos://").stripSuffix("/")
+    }
+
```

Comment:
This method doesn't belong in this class. It should be `protected` without an implementation and subclasses should override it with the appropriate logic